src/gg.regression.unity.bots/Runtime/Scripts/Validation/RGValidationScript.cs

```diff
@@ -0,0 +1,251 @@
+                         currentStatus == SegmentValidationStatus.PASSED)
+                {
+                    // Debug.LogError($"Validation method {validator.Method.Name} is now passing.");
+                    currentValidator.Status = SegmentValidationStatus.PASSED;
+                    OnValidationsUpdated?.Invoke();
+                } 
+                else if (currentValidator.Mode == ValidationMode.ONCE_TRUE_ALWAYS_TRUE &&
+                         currentValidator.Status == SegmentValidationStatus.UNKNOWN &&
+                         currentStatus == SegmentValidationStatus.FAILED)
+                {
+                    // Debug.LogError($"Validation method {validator.Method.Name} has now failed.");
+                    currentValidator.Status = SegmentValidationStatus.FAILED;
+                    OnValidationsUpdated?.Invoke();
+                }
+
+            }
+
+            frame++;
+        }
+
+        public void PauseValidation()
+        {
+            _isPaused = true;
+        }
+
+        public void UnPauseValidation()
+        {
+            _isPaused = false;
+        }
+        
+        public void StopValidations()
+        {
+            // First, make sure RGValidateBehaviour marks the final results as pass or fail based on the desired
+            // conditions.
+            foreach (var validator in Validators)
+            {
+                if (validator.Mode == ValidationMode.ALWAYS_TRUE && validator.Status == SegmentValidationStatus.UNKNOWN)
+                {
+                    //Debug.LogError($"Validation method {validator.Method.Name} should always pass, but it never did.");
+                    validator.Status = SegmentValidationStatus.FAILED;
+                    OnValidationsUpdated?.Invoke();
+                }
+                else if (validator.Mode == ValidationMode.NEVER_TRUE && validator.Status == SegmentValidationStatus.UNKNOWN)
+                {
+                    //Debug.LogError($"Validation method {validator.Method.Name} should never pass, and it never did!");
+                    validator.Status = SegmentValidationStatus.PASSED;
+                    OnValidationsUpdated?.Invoke();
+                }
+                else if (validator.Mode == ValidationMode.EVENTUALLY_TRUE && validator.Status is SegmentValidationStatus.UNKNOWN or SegmentValidationStatus.FAILED)
+                {
+                    //Debug.LogError($"Validation method {validator.Method.Name} should eventually pass, but it never did.");
+                    validator.Status = SegmentValidationStatus.FAILED;
+                    OnValidationsUpdated?.Invoke();
+                }
+                else if (validator.Mode == ValidationMode.ONCE_TRUE_ALWAYS_TRUE && validator.Status == SegmentValidationStatus.UNKNOWN)
+                {
+                    //Debug.LogError($"Validation method {validator.Method.Name} was never passed or failed, so it is technically passed.");
```

Comment:
This is counter intuitive to me. I would have expected this to fail. I would think: "If it never passed or failed, it technically didn't pass."

Can you explain it more?

Reply:
Yeah this is a good philosophical point... I could be swayed either way. I suppose you are right that in these cases, the user would expect that if it never turned true, they expected it to since they made this test. Going to make this a failure instead.